### ISSUE TITLE
Encryption fails with perl-5.22.1 on HP-UX-ia64

### ISSUE THREAD

**Issue by Tux** [doy/spreadsheet-parsexlsx#50]
```
This is perl 5, version 22, subversion 1 (v5.22.1) built for IA64.ARCHREV_0-thread-multi-LP64-ld

$ prove -vwb t/encryption.t
t/encryption.t .. Unsupported encryption algorithm: 0x6e3d2231 at /opt/perl64/.cpan/build/Spreadsheet-ParseXLSX-0.20-LpkYOW/blib/lib/Spreadsheet/ParseXLSX/Decryptor.pm line 99.
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run

Test Summary Report
-------------------
t/encryption.t (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
Files=1, Tests=0,  1 wallclock secs ( 0.08 usr  0.01 sys +  0.72 cusr  0.08 csys =  0.89 CPU)
Result: FAIL
```

**Comment by Tux** [comment]
Any news here?

**Comment by mazurin** [comment]
#51 not fixed this bug?

**Comment by Tux** [comment]
Probably, but it is not merged and not referenced *here*, so I overlooked.
Any chance on a new release?

**Comment by mazurin** [comment]
> Any chance on a new release?

No, sorry. I haven't enough time now.

**Comment by doy** [comment]
I'll review this in the next couple of days.

**Comment by doy** [comment]
Optimistically closing this out in the hopes that #51 fixes things. Feel free to reopen if not!

**Comment by doy** [comment]
@Tux I've released version 0.21-TRIAL - can you test it out and see if it works? I'm curious if this issue was actually fixed, but I also did a decent bit of refactoring to fix other bugs, and I'd like to make sure there weren't other regressions introduced as well (in case my test suite isn't comprehensive enough).

**Comment by Tux** [comment]
When testing 0.21 with Spreadsheet::Read, I now get
```
t/605_perc.t .... 1/?
#   Failed test 'Type B1 percentage'
#   at t/605_perc.t line 22.
#          got: 'numeric'
#     expected: 'percentage'
```

**Comment by Tux** [comment]
The tests take ages (15 minutes!), esp bug-32 on PA-RISC takes overly long
bug-57 also takes rather long
```
10 minutes:
HP-UX 11.31/64 U  rx2660/64 Itanium 2 9100/1710(2) ia64
This is perl 5, version 14, subversion 2 (v5.14.2) built for IA64.ARCHREV_0-LP64-ld
All tests successful.
Files=32, Tests=978, 571 wallclock secs ( 0.33 usr  0.13 sys + 472.30 cusr  3.27 csys = 476.03 CPU)
Result: PASS

15 minutes:
HP-UX 11.23/64 U  rx1620/64 Itanium 2/1600(2) ia64
This is perl 5, version 14, subversion 2 (v5.14.2) built for IA64.ARCHREV_0-LP64-ld
All tests successful.
Files=32, Tests=978, 845 wallclock secs ( 0.50 usr  0.19 sys + 706.75 cusr  4.32 csys = 711.76 CPU)
Result: PASS

1 hour and 25 minutes:
HP-UX 11.11/64 U  rp3410/64 PA8900/800 pa-2.0
This is perl 5, version 14, subversion 2 (v5.14.2) built for PA-RISC2.0-LP64
All tests successful.
Files=32, Tests=978, 5107 wallclock secs ( 0.89 usr  0.15 sys + 1552.78 cusr  4.05 csys = 1557.87 CPU)
Result: PASS
```

**Comment by doy** [comment]
I fixed the number format regression, and I sped up parsing a bunch (and released 0.22-TRIAL). Let me know if that's sufficient - I'm happy to move the slow tests to author-only tests or whatever if they're still a problem.

**Comment by Tux** [comment]
git checkout now passes again on Spreadsheet::Read
git checkout still passes on HP-UX ia64
11.31: PASS Files=32, Tests=978, 132 wallclock secs
11.23: PASS Files=32, Tests=978, 194 wallclock secs
11.11: PASS Files=32, Tests=978, 1072 wallclock secs
slow tests are not "slow" anymore on ia64, but they are still very slow on PA-2, but 1072 secs is much better than 5107 (18 minutes vs 85 minutes).

I just checked: you *do* have a HP-UX PA-RISC account with me :)
last used: Thu 05 Jul 2012 11:30:41
